src/components/Onboarding/utils/progress-utils.ts

```diff
@@ -1,28 +1,35 @@
 import { getNetworkType } from '@utils/web3-utils'
 import { dataURLtoFile, textToFile } from '@utils/kit-utils'
 
-const getStorageKey = (account, chainId) =>
+const getStorageKey = (account: string, chainId: number) =>
   `onboarding:${getNetworkType(chainId)}:${account}`
 
-export const getItem = (account, chainId) => {
+export const getItem = (account: string, chainId: number) => {
   const item = window.localStorage.getItem(getStorageKey(account, chainId))
   return item ? JSON.parse(item) : null
 }
 
-export const removeItem = account => {
-  window.localStorage.removeItem(getStorageKey(account))
+export const removeItem = (account: string, chainId: number) => {
+  window.localStorage.removeItem(getStorageKey(account, chainId))
 }
 
-export const setItem = (account, chainId, item) => {
-  window.localStorage.setItem(
-    getStorageKey(account, chainId),
-    JSON.stringify(item)
-  )
+export const setItem = (account: string, chainId: number, item: any) => {
+  try {
+    window.localStorage.setItem(
+      getStorageKey(account, chainId),
+      JSON.stringify(item)
+    )
+  } catch (error) {
+    throw new Error('Disk space exceed. Clear your cache.')
+  }
```

Comment:
More info on PR description - https://github.com/1Hive/gardens-ui/pull/789

Reply:
I wonder if a good place to clear the cache would be when the user clicks that don't want to continue from previous progress:

https://user-images.githubusercontent.com/9082013/150039306-445d1ecd-d707-4933-bd16-2ebf08c50f71.mov

Reply:
Ohh!! Nice catch! Yah new PR for that could be great and it easy fix will avoid full localstorage mainly in users do having tests.